src/main/java/org/spongepowered/api/text/title/Title.java

```diff
@@ -22,74 +22,116 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-
 package org.spongepowered.api.text.title;
 
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
 import com.google.common.base.Optional;
-import org.spongepowered.api.text.message.Message;
+import org.spongepowered.api.text.Text;
+
+import javax.annotation.Nullable;
 
 /**
  * Represents an immutable configuration for an in-game title. Instances of this
  * interface can be created through the {@link TitleBuilder} by calling
  * {@link Titles#builder()}.
  *
- * <p>
- * All properties of a title are optional - if they are not set it will use the
- * current default values from the client.
- * </p>
+ * <p>All properties of a title are optional - if they are not set it will use
+ * the current default values from the client.</p>
  */
-public interface Title {
+public class Title {
+
+    protected final Optional<Text> title;
+    protected final Optional<Text> subtitle;
+    protected final Optional<Integer> fadeIn;
+    protected final Optional<Integer> stay;
+    protected final Optional<Integer> fadeOut;
+    protected final boolean clear;
+    protected final boolean reset;
+
+    Title() {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (64/60 (106%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext7.iad.github.net ([192.30.252.198]:36462 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.84)
	(envelope-from <noreply@github.com>)
	id 1YNeZl-0003Xn-Sx
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:36:18 +0000
Date: Tue, 17 Feb 2015 01:36:17 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1424165777;
	bh=XB4JB92MLqoHDfE6SnZacHcVzs/bnzaC59rPGnZ9iJw=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=SnHVwtNie9TytBbwQnyuGqTc54JrPNco8rY/vOSFV+etsjnG5c74SE1KpHuK/LQbf
	 lLql9BHvMsybixHc4JtkITgclBvvdLzkRvNv+iWRC4Z7h5dQTvdA0IhBRfeF6/Dwq8
	 cKZ7JEJRG7mC8GpfwktNw7uXubMhieBce3fNkj0w=
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803079@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30b9125fa6_9293fd2b84ed2a099511d";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bf9b3ac7d5508229009ec12b43bff0021bcd25648592cf0000000110facd9192a170ce017a7707@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAv9UD4_M7FgiqZHb6MF_3JRwgjR0Cks5nswMRgaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk


----==_mimepart_54e30b9125fa6_9293fd2b84ed2a099511d
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>   */
> -public interface Title {
> +public class Title {
> +
> +    protected final Optional<Text> title;
> +    protected final Optional<Text> subtitle;
> +    protected final Optional<Integer> fadeIn;
> +    protected final Optional<Integer> stay;
> +    protected final Optional<Integer> fadeOut;
> +    protected final boolean clear;
> +    protected final boolean reset;
> +
> +    Title() {

why package-local

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803079
----==_mimepart_54e30b9125fa6_9293fd2b84ed2a099511d
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803079">src/main/java/org/spongepowered/api/text/title/Title.java</a>:</p>
<pre style='color:#555'>&gt;   */
&gt; -public interface Title {
&gt; +public class Title {
&gt; +
&gt; +    protected final Optional&lt;Text&gt; title;
&gt; +    protected final Optional&lt;Text&gt; subtitle;
&gt; +    protected final Optional&lt;Integer&gt; fadeIn;
&gt; +    protected final Optional&lt;Integer&gt; stay;
&gt; +    protected final Optional&lt;Integer&gt; fadeOut;
&gt; +    protected final boolean clear;
&gt; +    protected final boolean reset;
&gt; +
&gt; +    Title() {
</pre>
<p>why package-local</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803079">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv-n4j3rwyMlucM9isCzWohxnpwQOks5nswMRgaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803079"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30b9125fa6_9293fd2b84ed2a099511d--